modules/ROOT/pages/header-inject-remove-task.adoc

```diff
@@ -1,54 +1,76 @@
-= Header Injection and Removal
+= Header Injection 
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
 [width="100%", cols="5,15"]
 |===
->s| Policy Name | Header injection/Header removal
->s|Summary      | Adds/removes headers to a request or a response
+>s| Policy Name | Header Injection
+>s|Summary      | Adds headers to a request or a response
 >s|Category | Transformation
 >s| First Mule version available | 4.1.0
 .1+>.^s| Returned Status Codes
-|
+
+[%autowidth.spread,cols="a,a"]
+!===
+.1+>.^! 500 ! Configuration includes an expression that was not evaluated.
+! The provided expression in the configuration was not validated. +
+!===
+
 |===
-    
-The Header Injection policy adds headers to the request/response of the message. The Header Removal policy removes all listed headers.
+
+The Header Injection policy adds HTTP headers to the request or response of a message. When you configure this policy for your API, you must specify an inbound and outbound map of the headers that you want to add in the message processing in the form of a key-value pair:
 
 image::header-inject-remove-diagram.png[]
 
+You can optionally include DataWeave expressions for Mule 4.0 and Mule runtime engine (Mule) expressions for Mule 3.0 and in the value of the header:
+
+`“#[attributes.requestPath]”`
 
-== Applying Header Injection Policy
+In this example, all the headers matching the configured expression are injected to the message.
```

Comment:
1. The policy name in the UI is called Header Injection, so I want to use the term inject rather than add.
2. Explained the whole process to you on the call.